cardinal-sdk/src/commonTest/kotlin/com/icure/cardinal/sdk/crypto/KeylessPatientUserTest.kt

```diff
@@ -0,0 +1,314 @@
+				object : CryptoStrategies {
+					override suspend fun generateNewKeyForDataOwner(
+						self: DataOwnerWithType,
+						cryptoPrimitives: CryptoService,
+					): CryptoStrategies.KeyGenerationRequestResult {
+						return CryptoStrategies.KeyGenerationRequestResult.Keyless
+					}
+				},
+			)
+
+			// Before injecting exchange data, retrieval should fail
+			runCatching {
+				newPatientApi.calendarItem.getCalendarItem(ci.id)
+			}.isFailure shouldBe true
+
+			// Inject exchange data (simulate what the TS test does)
+			newPatientApi.crypto.injectExchangeData(
+				patientDetails.groupId,
+				listOf(
+					ExchangeDataInjectionDetails(
+						exchangeDataId = exchangeData.exchangeDataId,
+						exchangeKey = exchangeData.exchangeKey,
+						accessControlSecret = exchangeData.accessControlSecret,
+						sharedSignatureKey = exchangeData.sharedSignatureKey,
+						verified = false,
+					),
+				),
+				false,
+			)
+
+			// After injection, retrieval should succeed
+			val retrievedByNewApi = newPatientApi.calendarItem.getCalendarItem(ci.id)
+			retrievedByNewApi.shouldNotBeNull()
+			retrievedByNewApi.title shouldBe ci.title
+			newPatientApi.calendarItem.decryptPatientIdOf(retrievedByNewApi).map { it.entityId } shouldBe listOf(patient.id)
+		}
+
+		"Keyless SDks do not support the re-encryption of injected exchange data" {
+			// Arrange: initialize test data
+			val (patientApi, exchangeData) = getPatientApi()
+			val patientUser = patientApi.user.getCurrentUser()
+			val patient = patientApi.patient.encrypted.getPatient(patientUser.dataOwnerId).shouldNotBeNull()
+			val ci = createCalendarItem(patientApi, patient, patientUser)
+
+			val newCryptoStrategies = object : CryptoStrategies {
+				override suspend fun generateNewKeyForDataOwner(
+					self: DataOwnerWithType,
+					cryptoPrimitives: CryptoService,
+				): CryptoStrategies.KeyGenerationRequestResult {
+					// Simulate a new key by returning Keyless (or adapt if keyful is needed)
+					return CryptoStrategies.KeyGenerationRequestResult.Keyless
+				}
+			}
+
+			val newPatientApiWithKeyAndInjected = patientDetails.api(
+				specJob,
+				object : CryptoStrategies {
+					override suspend fun generateNewKeyForDataOwner(
+						self: DataOwnerWithType,
+						cryptoPrimitives: CryptoService,
+					): CryptoStrategies.KeyGenerationRequestResult {
+						return CryptoStrategies.KeyGenerationRequestResult.Keyless
+					}
+				},
+			)
+
+			runCatching {
+				// Inject exchange data (simulate what the TS test does)
+				newPatientApiWithKeyAndInjected.crypto.injectExchangeData(
+					patientDetails.groupId,
+					listOf(
+						ExchangeDataInjectionDetails(
+							exchangeDataId = exchangeData.exchangeDataId,
+							exchangeKey = exchangeData.exchangeKey,
+							accessControlSecret = exchangeData.accessControlSecret,
+							sharedSignatureKey = exchangeData.sharedSignatureKey,
+							verified = false,
+						),
+					),
+					true,
+				)
+			}.isFailure shouldBe true
```

Comment:
What do you mean "// Inject exchange data (simulate what the TS test does)"

Again use should throw and check appropriate exception